src/mongocxx/exception/error_code.hpp

```diff
@@ -31,6 +31,92 @@ enum class error_code : std::int32_t {
     k_unknown_write_concern,
 };
 
+MONGOCXX_API const std::error_code& bson_invalid_error();
```

Comment:
It is possibly preferable to exposing the C driver error codes and error domains, because if the C++ driver were to stop being implemented in terms of the C driver, then we could just change the implementation of bson_invalid_error() and not break users' code. But it may not be worth it.